notifications-lambda/src/index.ts

```diff
@@ -4,34 +4,36 @@ import {
   pinboardSecretPromiseGetter,
   standardAwsConfig,
 } from "../../shared/awsIntegration";
-import { AttributeMap, Key } from "aws-sdk/clients/dynamodb";
+import { Key } from "aws-sdk/clients/dynamodb";
 import { Item, MyUser } from "../../shared/graphql/graphql";
 import { getEnvironmentVariableOrThrow } from "../../shared/environmentVariables";
 import { publicVapidKey } from "../../shared/constants";
-
-interface DynamoStreamRecord {
-  dynamodb: {
-    NewImage: AttributeMap;
-  };
-}
-
-interface DynamoStreamEvent {
-  Records: DynamoStreamRecord[];
-}
+import { DynamoDBStreamEvent } from "aws-lambda";
 
 const isUserMentioned = (item: Item, user: MyUser) =>
   item.mentions?.includes(user.email);
 
 const doesUserManuallyHavePinboardOpen = (item: Item, user: MyUser) =>
   user.manuallyOpenedPinboardIds?.includes(item.pinboardId);
 
-export const handler = async (event: DynamoStreamEvent) => {
+export const handler = async (event: DynamoDBStreamEvent) => {
   const dynamo = new AWS.DynamoDB.DocumentClient(standardAwsConfig);
   const usersTableName = getEnvironmentVariableOrThrow("usersTableName");
   const privateVapidKey = await pinboardSecretPromiseGetter(
     "notifications/privateVapidKey"
   );
 
+  const items: Item[] = event.Records.reduce(
+    (acc, record) =>
+      record.dynamodb?.NewImage
+        ? [
+            ...acc,
+            AWS.DynamoDB.Converter.unmarshall(record.dynamodb.NewImage) as Item,
+          ]
```

Comment:
the official type has `dynamodb` and `NewImage` as optional, so we now `reduce` to filter those out whilst also mapping to `Item`. We also do this once now, rather than for each matching user result (was wasteful before).